prow/plank/reconciler.go

```diff
@@ -239,22 +239,24 @@ func (r *reconciler) terminateDupes(pj *prowv1.ProwJob) error {
 		return fmt.Errorf("failed to list prowjobs: %v", err)
 	}
 
-	return pjutil.TerminateOlderJobs(r.pjClient, r.log, pjs.Items, func(toCancel prowv1.ProwJob) error {
-		client, ok := r.buildClients[pj.ClusterAlias()]
-		if !ok {
-			return fmt.Errorf("no client for cluster %q present", pj.ClusterAlias())
-		}
-		podToDelete := &corev1.Pod{
-			ObjectMeta: metav1.ObjectMeta{
-				Namespace: r.config().PodNamespace,
-				Name:      toCancel.Name,
-			},
-		}
-		if err := client.Delete(r.ctx, podToDelete); err != nil {
-			return fmt.Errorf("failed to delete pod %s/%s: %w", podToDelete.Namespace, podToDelete.Name, err)
-		}
-		return nil
-	})
+	return pjutil.TerminateOlderJobs(r.pjClient, r.log, pjs.Items, r.terminateDupesCleanup)
+}
+
+func (r *reconciler) terminateDupesCleanup(pj prowv1.ProwJob) error {
+	client, ok := r.buildClients[pj.ClusterAlias()]
+	if !ok {
+		return fmt.Errorf("no client for cluster %q present", pj.ClusterAlias())
+	}
+	podToDelete := &corev1.Pod{
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: r.config().PodNamespace,
+			Name:      pj.Name,
+		},
+	}
+	if err := client.Delete(r.ctx, podToDelete); err != nil && !kerrors.IsNotFound(err) {
```

Comment:
if it is already deleted would we want to info/debug/trace this?

Reply:
This is expected to happen sometimes if ppl push rapidly